src/processTree.ts

```diff
@@ -125,27 +125,25 @@ export function getProcesses(one: (pid: number, ppid: number, command: string, a
 
 		} else {	// linux
 
-			proc = spawn('/bin/ps', [ '-ax', '-o', 'pid,ppid,comm:20,command' ]);
-			proc.stdout?.setEncoding('utf8');
-			proc.stdout?.on('data', lines(line => {
-
-				const pid = Number(line.substr(0, 5));
-				const ppid = Number(line.substr(6, 5));
-				let command = line.substr(12, 20).trim();
-				let args = line.substr(33);
-
-				let pos = args.indexOf(command);
-				if (pos >= 0) {
-					pos = pos + command.length;
-					while (pos < args.length) {
-						if (args[pos] === ' ') {
-							break;
-						}
-						pos++;
-					}
-					command = args.substr(0, pos);
-					args = args.substr(pos + 1);
-				}
+			proc = spawn('/bin/ps', [ '-ax', '-o', 'pid:6,ppid:6,comm:20,command' ]);	// we specify the column width explicitly
+			proc.stdout.setEncoding('utf8');
+			proc.stdout.on('data', lines(line => {
+				
+				// the following substr arguments must match the column width specified for the "ps" command above
+				// regular substr is deprecated
+				const pid = Number(substr(line, 0, 6));
+				const ppid = Number(substr(line, 7, 6));
+				const shortName = substr(line, 14, 20).trim()
+				const fullCommand = substr(line, 35)
+
+				// binaries with spaces in path may not work
+				// possible solution to read directly from /proc
+				let pos = fullCommand.indexOf(shortName);
+				const commandEndPositionMaybe = fullCommand.indexOf(" ", pos + shortName.length)
+				const commandEndPosition = commandEndPositionMaybe < 0 ? fullCommand.length : commandEndPositionMaybe;
+
+				const command = fullCommand.substring(0, commandEndPosition)
+				const args = fullCommand.substring(commandEndPosition).trimStart()
```

Comment:
You should consider all possible corner cases as possible.

Suggested as below:
```
                let command = shortName;
                let args = fullCommand;
                const pos = fullCommand.indexOf(shortName);
                if (pos >= 0) {
                      const commandEndPositionMaybe = fullCommand.indexOf(" ", pos + shortName.length);
                      const commandEndPosition = commandEndPositionMaybe < 0 ? fullCommand.length : commandEndPositionMaybe;
                      command = fullCommand.substring(0, commandEndPosition)
				      args = fullCommand.substring(commandEndPosition).trimStart()
                }
```

Reply:
Although I could not even imagine how this case could ever happen and why not just skip such process, I have incorporated this change into code. Check newly pushed commit